teku/src/main/java/tech/pegasys/teku/cli/options/ValidatorKeysOptions.java

```diff
@@ -173,4 +169,50 @@ private Path convertToPath(final String option) {
     }
     return Path.of(option);
   }
+
+  static class PublicKeyLoader {
+    final ObjectMapper objectMapper;
+
+    public PublicKeyLoader() {
+      this(new ObjectMapper());
+    }
+
+    public PublicKeyLoader(final ObjectMapper objectMapper) {
+      this.objectMapper = objectMapper;
+    }
+
+    public List<BLSPublicKey> getPublicKeys(final List<String> publicKeys) {
+      if (publicKeys == null || publicKeys.isEmpty()) {
+        return Collections.emptyList();
+      }
+
+      try {
+        final Set<BLSPublicKey> blsPublicKeySet =
+            publicKeys.stream()
+                .filter(key -> !key.contains(":"))
+                .map(key -> BLSPublicKey.fromSSZBytes(Bytes.fromHexString(key)))
+                .collect(Collectors.toSet());
+        blsPublicKeySet.addAll(
+            publicKeys.stream()
+                .filter(key -> key.contains(":"))
+                .flatMap(this::readKeysFromUrl)
+                .collect(Collectors.toList()));
```

Comment:
Probably just do this in one pass:
```suggestion
        final Set<BLSPublicKey> blsPublicKeySet =
            publicKeys.stream()
                .filter(key -> !key.contains(":"))
                .flatMap(value -> key.contains(':') ? readKeysFromUrl(value) : BLSPublicKey.fromSSZBytes(Bytes.fromHexString(value)))
                .collect(Collectors.toSet());
```